storey/drivers.py

```diff
@@ -518,3 +518,137 @@ async def _get_item(self, container, table_path, key, attributes):
 
         return await self._v3io_client.kv.get(container, table_path, str(key), attribute_names=attributes,
                                               raise_for_status=v3io.aio.dataplane.RaiseForStatus.never)
+
+
+class SQLDriver(Driver):
+    """
+    Database connection to Sql data basw.
+    :param db_path: url string connection to sql database.
+    :param primary_key: the primary key of the table.
+    """
+
+    def __init__(
+        self,
+        primary_key: str,
+        db_path: str,
+        aggregation_attribute_prefix: str = "aggr_",
+        aggregation_time_attribute_prefix: str = "_",
+    ):
+        self._db_path = db_path
+        self._sql_connection = None
+        self._primary_key = primary_key
+        self._mtime_name = "$_mtime_"
+        self._storey_key = "storey_key"
+
+        self._aggregation_attribute_prefix = aggregation_attribute_prefix
+        self._aggregation_time_attribute_prefix = aggregation_time_attribute_prefix
+
+    def _lazy_init(self):
+        import sqlalchemy as db
+
+        if not self._sql_connection:
+            self._engine = db.create_engine(self._db_path)
+            self._sql_connection = self._engine.connect()
+
+    def table(self, table_path):
+        import sqlalchemy as db
+
+        metadata = db.MetaData()
+        return db.Table(
+            table_path[3:].split("/")[1],
+            metadata,
+            autoload=True,
+            autoload_with=self._engine,
+        )
+
+    async def _save_schema(self, container, table_path, schema):
+        self._lazy_init()
+        return None
+
+    async def _load_schema(self, container, table_path):
+        self._lazy_init()
+        return None
+
+    async def _save_key(
+        self, container, table_path, key, aggr_item, partitioned_by_key, additional_data
+    ):
+        import sqlalchemy as db
+
+        self._lazy_init()
+
+        table = self.table(table_path)
+        return_val = None
+        try:
+            return_val = self._sql_connection.execute(table.insert(), [additional_data], autocommit=True)
+
+        except db.exc.IntegrityError:
+            pass
+        await self.close()
+        return return_val
```

Comment:
Why return anything? `_save_key` isn't supposed to return anything, and the return value is never used.